eng/doc/DeveloperGuide.md

```diff
@@ -0,0 +1,84 @@
+# Developer Guide
+
+This document is a guide for developers who want to contribute to the Microsoft Go repository.
+It explains how to build the repository, how to work with the Go submodule, and how to use the different tools
+that help maintain the repository.
+
+## Setting up the repository
+
+### Step 0: Contributor License Agreement
+
+Most contributions require you to agree to a Contributor License Agreement (CLA) declaring that you have the right to, and actually do, grant us the rights to use your contribution.
+For details, visit https://cla.opensource.microsoft.com.
+
+### Step 1: Install a Go toolchain
+
+A preexisting Go toolchain is required to bootstrap the build process.
+You can use your system's package manager to install Go, or you can download it from the [official Go website](https://golang.org/dl/).
+The only requirement is that the Go version is high enough for the bootstrap process.
+If the version is too low, the bootstrap process will fail and ask you to install a newer version.
+
+### Step 2: Install the git-go-patch command
+
+The `git-go-patch` command is a tool that helps you manage the patches in the `go` submodule.
+
+To install the `git-go-patch` command, run the following command:
+
+```
+go install github.com/microsoft/go-infra/cmd/go-patch@latest
+```
+
+> [!NOTE]
+> Make sure `git-go-patch` is accessible in your shell's `PATH` variable.
+> You may need to add `$GOPATH/bin` to your `PATH`. Use `go env GOPATH` to locate it.
+
+Then, run the command to see the help documentation:
+
+```
+git go-patch -h
+```
+
+> [!NOTE]
+> `git` detects that our `git-go-patch` executable starts with `git-` and makes it available as `git go-patch`.
+> The program still works if you call it with its real name, but we think it's easier to remember and type something that looks like a `git` subcommand.
```

Comment:
I don't think this is important enough to note here.

Reply:
I'll remove the last sentence. I think the first sentence does add some value.